File: Problem_Set_02/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Function returns the grade level
int gradecalc(string text)
{
    // Count characters
    int length = strlen(text);

    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Loop to count letter, word, and sentence
    for (int i = 0; i < length; i++)
    {
        char c = text[i];

        // character is A-Z or a-z
        if (isalpha(c))
        {
            letters++;
        }

        // space means word ended
        if (c == ' ')
        {
            words++;
        }

        // ., !, and ?  end of sentence
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    words++;

    // average number of letters
    float L = ((float) letters / (float) words) * 100;

    // average number of sentence
    float S = ((float) sentences / (float) words) * 100;

    // Coleman-Liau index formula
    float index = 0.0588 * L - 0.296 * S - 15.8;

    int idx = round(index);

    return idx;
}

int main(void)
{

    string text = get_string("Text: ");

    // Compute the grade level
    int grade = gradecalc(text);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: Problem_Set_06/sentimental/readability/readability.py
from cs50 import get_string


text = get_string("Text: ").strip()

letter_count = 0
word_count = 1
sentence_count = 0

for char in text:
    if char.isalpha():
        letter_count += 1
    elif char.isspace():
        word_count += 1
    elif char in ".!?":
        sentence_count += 1

letters_per_100_words = (letter_count / word_count) * 100
sentences_per_100_words = (sentence_count / word_count) * 100

grade_level = round(
    0.0588 * letters_per_100_words
    - 0.29 * sentences_per_100_words
    - 15.8
)

if grade_level >= 16:
    print("Grade 16+")
elif grade_level >= 1:
    print(f"Grade {grade_level}")
else:
    print("Before Grade 1")
